### ISSUE TITLE
Long journeys likely counted multiple times

### ISSUE THREAD

**Issue by hsl-petrhaj** [HSLdevcom/helmet-model-system#551]
I know there is likely only a small share of these in the model, but it seems like those journeys that do not fit into one-hour window are likely counted multiple times. Probably the best way to fix that would be to divide the amount of journeys by their travel time in hours. For example two-hour public transport journeys from Hämeenlinna to Porvoo would be divided by two. This might be even more relevant for external traffic, if this has not been taken into account.

I ran some small tests in Emme and nothing suggests that the assignment directly handles that, perhaps except for the space-time assignment.
@zptro

**Comment by zptro** [comment]
I am not sure I understand exactly what you are referring to, but I will try to explain how I see this. When doing model analyses we have two types of results; (1) matrix-level results and (2) link-level results. 
1. Consumer surplus (user benefits) is and should always be calculated on the matrix level, and on the matrix level all journeys occur exactly once. On the matrix level we often have problems with defining the peak hour for long journeys (is it based on departure time, arrival time or something else?), but that is something we simply have to deal with. The number of journeys for a one-hour window is however a well defined concept on the matrix level.
2. External effects are calculated on the link level. It may seem strange that a two-hour journey causes traffic on all links along the route during a single-hour assignment, but in this context we cannot think of assigned traffic flow as individual journeys propagating through the network, but rather as what is actually observed on the link; in your example, what is observed on links near Hämeenlinna is journeys starting ten minutes ago (counted in a one-hour time window), while traffic observed near Porvoo consists of trips that started two hours ago (also counted in a one-hour window). Hence they are not actually the same journeys.

To summarize, static assignment is not actually dealing with journeys, but with traffic. A large share of link-level traffic comes from journeys starting or ending outside of the analyzed time window, but traffic can still be counted for that precise time window.

**Comment by zptro** [comment]
> Thanks for the explanation.
> 
> > To summarize, static assignment is not actually dealing with journeys, but with traffic.
> 
> This is quite important to know. So O-D pair represents 20 "traffic", not 20 journeys. Is the 20 "traffic" then the average of the counts along the route? I drew some space-time diagram for myself and it feels hard to believe that a single OD pair would cause the same amount of traffic across the whole two-hour journey. But average would make some sense.

I stated that *assignment* deals with traffic only. The concept of an O-D pair is however not really an assignment thing, but a travel demand thing (i.e., a matrix-level concept). Static assignment transforms O-D matrices into link flows (i..e, traffic), assuming demand to be constant before, during and after the assigned time window. That is a rather strong assumption of course, but for long journeys it may not be crucial.
